### DISCUSSION TITLE
Prisma enum value is automatically set to UPPERCASE format in API response

### DISCUSSION THREAD

**Discussion by daiho-nfq** [prisma/prisma#17062] [Q&A]
I have an enum named `UserRole`

enum UserRole {
    PLAYER                  @map("player")
    SUPER_PLAYER      @map("super-player")
    MASTER_PLAYER   @map("master-player")
}

then I use the enum in model User

model User {
     id String 
     name String
     ...
     role UserRole @default(PLAYER)
}

Everything works perfectly until I create api to get all users. The value of `role` in the response is set to `UPPERCASE`, I check in the database, it is stored as `lowercase` format.

As far as I research, `@map` is just applied only in the database. Does anyone get the same issue?

I am using NestJS framework with mysql

**Comment by daiho-nfq** [comment]
@nurul3101 Hi, Could you please help me with this issue?

**Comment by nurul3101** [comment]
Hello @daiho-nfq 👋 
Your observation is correct, the mapped value is only applied to the database.

I believe your API is reading data from the database? In that case, the response should have the same value as stored in the database. Can you elaborate on what is your exact use case and what you would like to achieve?

**Reply by daiho-nfq** [reply]
Hi @nurul3101 👋 
I have an enum UserRole

enum UserRole {
`PLAYER @map("player")`
`SUPER_PLAYER @map("super-player")`
`MASTER_PLAYER @map("master-player")`
}

and model User
model User {
`id String`
`name String`
...
`role UserRole @default(PLAYER)`
}


I have 10 user records in the database

- in the database, the value of `role` is lowercase (player, super-player, master-player) ->  kebab-case
- in the api response `getAll()`,  the value of `role` is UPPERCASE (PLAYER, SUPER_PLAYER, MASTER_PLAYER) -> snake_case

The thing I want to achieve is that in the API response, is there any solution to get role value as `lowercase` mapped value (player, super-player, master-player) ->  kebab-case

**Reply by nurul3101** [reply]
Can you share how you are querying records in `getAll()` function? Are you reading the records from the database?

**Reply by daiho-nfq** [reply]
very simple query 

In UserService
`async getAll() {
    return this.prisma.user.findMany();
  }`

in UserController
@Get()
`list() {
    return this.usersService.getAll();
  }`

**Reply by nurul3101** [reply]
Ah, I get it now. You would need to manually map the uppercase values to lowercase values and return lowercase values when an uppercase value is received from the query. We have a related GitHub Issue here
- https://github.com/prisma/prisma/issues/8446

**Reply by daiho-nfq** [reply]
@nurul3101 oh thank you. I think I will use UPPERCASE for enum instead of `manual mapping`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
